src/pint/models/noise_model.py

```diff
@@ -491,41 +490,51 @@ def __init__(
                 units="",
                 aliases=[],
                 description="Amplitude of powerlaw DM noise in tempo2 format",
-                convert_tcb2tdb=False,
+                convert_tcb2tdb=True,
+                tcb2tdb_scale_factor=1,
             )
         )
         self.add_param(
             floatParameter(
                 name="TNDMGAM",
                 units="",
                 aliases=[],
-                description="Spectral index of powerlaw " "DM noise in tempo2 format",
-                convert_tcb2tdb=False,
+                description="Spectral index of powerlaw DM noise in tempo2 format",
+                convert_tcb2tdb=True,
+                tcb2tdb_scale_factor=1,
             )
         )
         self.add_param(
-            floatParameter(
+            intParameter(
                 name="TNDMC",
                 units="",
                 aliases=[],
                 description="Number of DM noise frequencies.",
-                convert_tcb2tdb=False,
             )
         )
         self.add_param(
-            floatParameter(
+            intParameter(
                 name="TNDMFLOG",
                 units="",
                 description="Number of logarithmically spaced DM noise frequencies in the basis.",
-                convert_tcb2tdb=False,
             )
         )
         self.add_param(
             floatParameter(
                 name="TNDMFLOG_FACTOR",
                 units="",
-                description="Factor of the log-spaced DM frequencies (2 -> [1/8,1/4,1/2,...])",
-                convert_tcb2tdb=False,
+                description="Factor of the log-spaced DM frequencies (2 -> [1/8, 1/4, 1/2, ...]).",
+                convert_tcb2tdb=True,
+                tcb2tdb_scale_factor=1,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNDMTSPAN",
+                units="year",
+                description="Time span corresponding to the fundamental frequency of the DM noise Fourier series.",
```

Comment:
Should this comment have "Tspan if `None`" or similar?

Reply:
added.